gems/aws-sdk-core/lib/seahorse/client/plugins/h2.rb

```diff
@@ -28,11 +28,14 @@ class H2 < Plugin
         DOCS
 
         # SSL Context
-        option(:ssl_ca_bundle, default: nil, doc_type: String, docstring: <<-DOCS)
+        option(:ssl_ca_bundle, default: nil, doc_type: String, docstring: <<-DOCS) do |cfg|
 Full path to the SSL certificate authority bundle file that should be used when
 verifying peer certificates. If you do not pass `:ssl_ca_directory` or `:ssl_ca_bundle`
 the system default will be used if available.
         DOCS
+          ENV['AWS_CA_BUNDLE'] ||
```

Comment:
My understanding is that it's better (good practice) to use `fetch` method for such logic. Something like:

```
ENV.fetch('AWS_CA_BUNDLE', Aws.shared_config.ca_bundle(profile: cfg.profile)) if cfg.respond_to?(:profile)
```

Reply:
I don't think that can be done here because we can only fall back to the shared config if the config responds to profile.

Reply:
I'm still using `if cfg.respond_to?(:profile)` which guarantees that `we can only fall back to the shared config if the config responds to profile`. It's just the way we fall back, instead of using `||` I was encouraging the the PR author to use `Hash#fetch`.

Reply:
I'd agree in general its preferred to use `fetch` for fallback ( since cases where the value are set to false will fallback with `||`).   However I think you're code is incorrect, and needs to be:
```ruby
 ENV.fetch('AWS_CA_BUNDLE', (Aws.shared_config.ca_bundle(profile: cfg.profile) if cfg.respond_to?(:profile))) # parens around the default value w/ if
```

However, for ENV the values are restricted to strings so the issues with `||` don't apply.  The rest of the code uses `||` with ENV so I'd prefer to keep that style consistent here.

Reply:
Yes agreed. When I meant "I don't think that can be done here" it was because the if statement was outside of ENV.fetch. It adds extra complexity here in this case.